### ISSUE TITLE
Small issue with Atmosphere on Websphere Liberty when shutting down the server.

### ISSUE THREAD

**Issue by seamusmac** [Atmosphere/atmosphere#1974]
I have recently upgraded to the latest Atmosphere version, 2.3.0.

I have been noticing this error, whenever I shutdown the server. Its not a big issue, but thought I would let you know, May be a very simple fix. The stacktrace is below:

```
[AUDIT   ] WTRN0105I: The transaction service has shutdown successfully with no transactions requiring recovery.
[AUDIT   ] CWWKE0036I: The server baseServer stopped after 55 minutes, 44.679 seconds.
Exception in thread "Thread-27" java.lang.RuntimeException: SESN0007E: There was an attempt to access a session while the application server was stopping. 
	at com.ibm.ws.session.store.memory.MemoryStore.throwException(MemoryStore.java:758)
	at com.ibm.ws.session.store.memory.MemoryStore.getSession(MemoryStore.java:242)
	at com.ibm.ws.session.store.memory.MemoryStore.getSession(MemoryStore.java:814)
	at com.ibm.ws.session.SessionManager.getSessionFromStore(SessionManager.java:466)
	at com.ibm.ws.session.SessionManager.getSession(SessionManager.java:444)
	at com.ibm.ws.session.SessionManager.getSession(SessionManager.java:431)
	at com.ibm.ws.session.SessionManager.getSession(SessionManager.java:678)
	at com.ibm.ws.webcontainer.session.impl.HttpSessionContextImpl.getIHttpSession(HttpSessionContextImpl.java:473)
	at com.ibm.ws.webcontainer.session.impl.HttpSessionContextImpl.getIHttpSession(HttpSessionContextImpl.java:637)
	at com.ibm.ws.webcontainer.srt.SRTRequestContext.getSession(SRTRequestContext.java:105)
	at com.ibm.ws.webcontainer.srt.SRTServletRequest.getSession(SRTServletRequest.java:2161)
	at javax.servlet.http.HttpServletRequestWrapper.getSession(HttpServletRequestWrapper.java:238)
	at org.apache.shiro.web.servlet.ShiroHttpServletRequest.getSession(ShiroHttpServletRequest.java:143)
	at org.atmosphere.cpr.AtmosphereRequest.getSession(AtmosphereRequest.java:796)
	at org.atmosphere.cpr.SessionTimeoutSupport.restoreTimeout(SessionTimeoutSupport.java:69)
	at org.atmosphere.cpr.AtmosphereResourceImpl.cancel(AtmosphereResourceImpl.java:822)
	at org.atmosphere.cpr.AtmosphereResourceImpl.close(AtmosphereResourceImpl.java:903)
	at org.atmosphere.cpr.AtmosphereFramework.closeAtmosphereResource(AtmosphereFramework.java:1757)
	at org.atmosphere.cpr.AtmosphereFramework.destroy(AtmosphereFramework.java:1770)
	at org.atmosphere.cpr.AtmosphereFramework$4.run(AtmosphereFramework.java:925)
```

**Comment by seamusmac** [comment]
I have been testing this with Atmosphere 2.4.0, and have found 2 small issues related to this. Do you want me to submit a PR that fixes it.

**Comment by jfarcand** [comment]
@seamusmac +1

**Comment by seamusmac** [comment]
Testing right now, also found another issue in the AtmosphereFramework class where the shutdown hook is made null before it is removed.

**Comment by seamusmac** [comment]
I had to add the catch to endAsyncContext cos it was being called 2 times, due to the AtmosphereFramework.closeAtmosphereResource() method.

It calls: ar.resume().close();

1. resume -- >> Servlet30CometSupport.action > Servlet30CometSupport.endAsyncContext 
2. close  -- >> ar.cancel > Servlet30CometSupport.complete > Servlet30CometSupport.endAsyncContext

**Comment by seamusmac** [comment]
@jfarcand  what do you think, should complete be called 2 times on shutdown? Is the PR ok?

**Comment by jfarcand** [comment]
@seamusmac No two times must be fixed. Is your pull fixing this? If not, pull request :-)

**Comment by seamusmac** [comment]
@jfarcand 2 times is fixed now, also remove the last catch I added in previous PR as it is not needed now.

**Comment by jfarcand** [comment]
@seamusmac Too late I've released `2.4.0.1` :-)

**Comment by seamusmac** [comment]
@jfarcand thanks for that, I have updated to last version 2.4.0.1 and all works well :+1:
